File: extension/test-samples/logging-example.ts
// TypeScript logging example for PII detection

function processUser(firstName: string, lastName: string) {
  // These SHOULD trigger pii-logging-unmasked warnings
  console.log("User:", firstName, lastName);
  console.warn(`Processing ${firstName} ${lastName}`);
  console.error("Failed for user: " + firstName);

  // These should NOT trigger logging warnings (masked)
  console.log("User:", maskHelper.mask(firstName));
  console.log("User: [REDACTED]");

  // These should NOT trigger logging warnings (not in log context)
  const displayName = `${firstName} ${lastName}`;
  const user = { firstName, lastName };
  return user;
}

File: extension/test-samples/LoggingExample.cs
using Microsoft.Extensions.Logging;

namespace SampleApp.Services
{
    public class UserService
    {
        private readonly ILogger<UserService> _logger;

        public void ProcessUser(string firstName, string lastName)
        {
            // These SHOULD trigger pii-logging-unmasked warnings
            Console.WriteLine($"Processing user: {firstName} {lastName}");
            _logger.LogInformation("User first name: " + firstName);
            _logger.LogWarning($"User {lastName} has issues");
            Debug.WriteLine("Name: " + firstName);

            // These should NOT trigger logging warnings (masked)
            _logger.LogInformation("User: " + MaskHelper.Mask(firstName));
            Console.WriteLine($"User: {DataMasker.Redact(lastName)}");
            _logger.LogInformation("User: [REDACTED]");

            // These should NOT trigger logging warnings (not in log context)
            var displayName = firstName + " " + lastName;
            var user = new { FirstName = firstName };

            // Multi-line logging — SHOULD trigger warning
            _logger.LogInformation(
                "User details: {0} {1}",
                firstName,
                lastName
            );
        }
    }
}
